bowlingGame.js

```diff
@@ -0,0 +1,65 @@
+class BowlingGame {
+  constructor(){
+    this.rolls = [];
+    this.currentFrame = 1;
+  };
+  roll(number){
+    if(number > 10){
+      throw "Invalid Roll";
+    }
+    else {
+      this.rolls.push( { value: number, frame: this.currentFrame } );
+    };
+    
+  };
+  // [10, 5, 5, 10, 5, 3]
+  score(){
+    let totalScore = 0;
+    let currentRoll = 0;
+    while(currentRoll < this.rolls.length){
+      if(this.currentFrame == 10){
+        totalScore += this.sumTheRest(currentRoll);
+        break;
+      }
+      else if(this.frameIsStrike(currentRoll)){       
+        totalScore += this.bonusScoring(currentRoll);
+        currentRoll += 1;
+        this.currentFrame += 1
+      }
+      else if(this.frameIsSpare(currentRoll)){
+        totalScore += this.bonusScoring(currentRoll);
+        currentRoll += 2;
+        this.currentFrame += 1
+      }
+      else {
+        totalScore += this.rolls[currentRoll].value + this.rolls[currentRoll + 1].value;
+        currentRoll += 2;
+        this.currentFrame += 1
+      };
+      
+    }
+    return totalScore;
+  };
+
+  sumTheRest(currentRoll){
+    let sum = 0;
+    this.rolls.slice(currentRoll).forEach(roll => {
+      sum += roll.value
+    });
+    return sum;
+  };
+  frameIsSpare(currentRoll){
+    return this.rolls[currentRoll].value + this.rolls[currentRoll + 1].value == 10;
```

Comment:
Let's say that the second roll in a frame was 5 and the first roll in the next frame was 5 also. Would the frameIsSpare method incorrectly think this is a somehow a spare?
I may be wrong but the frameIsSpare method doesn't seem to reference the frame that the 'currentRoll' is in so I think it may get the mentioned scenario wrong.